pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -682,45 +658,125 @@ func FindPrimaryDNSProvider(providers []gardencorev1beta1.DNSProvider) *gardenco
 // VersionPredicate is a function that evaluates a condition on the given versions.
 type VersionPredicate func(expirableVersion gardencorev1beta1.ExpirableVersion, version *semver.Version) (bool, error)
 
-// GetKubernetesVersionForPatchUpdate finds the latest Kubernetes patch version for its minor version in the <cloudProfile> compared
-// to the given <currentVersion>. Preview and expired versions do not qualify for the kubernetes patch update. In case it does not find a newer patch version, it returns false. Otherwise,
-// true and the found version will be returned.
-func GetKubernetesVersionForPatchUpdate(cloudProfile *gardencorev1beta1.CloudProfile, currentVersion string) (bool, string, error) {
+// GetExpirableVersionsForMachineImageVersions returns a slice of expirable versions from a slice of machine image versions
+func GetExpirableVersionsForMachineImageVersions(versions []gardencorev1beta1.MachineImageVersion) []gardencorev1beta1.ExpirableVersion {
+	var expirableVersions []gardencorev1beta1.ExpirableVersion
+	for _, v := range versions {
+		expirableVersions = append(expirableVersions, v.ExpirableVersion)
+	}
+	return expirableVersions
+}
+
+// GetLatestVersionForPatchAutoUpdate finds the latest patch version for a given <currentVersion> for the current minor version from a given slice of versions.
+// The current version, preview and expired versions do not qualify.
+// In case no newer patch version is found, returns false and an empty string. Otherwise, returns true and the found version.
+func GetLatestVersionForPatchAutoUpdate(versions []gardencorev1beta1.ExpirableVersion, currentVersion string) (bool, string, error) {
+	currentSemVerVersion, err := semver.NewVersion(currentVersion)
+	if err != nil {
+		return false, "", err
+	}
+
+	predicates := []VersionPredicate{FilterDifferentMajorMinorVersionAndLowerPatchVersionsOfSameMinor(*currentSemVerVersion)}
+
+	return getVersionForAutoUpdate(versions, currentSemVerVersion, predicates)
+}
+
+// GetLatestVersionForMinorAutoUpdate finds the latest minor with the latest patch version higher than a given <currentVersion> for the current major version from a given slice of versions.
+// The current version, preview and expired versions do not qualify.
+// In case no newer version is found, returns false and an empty string. Otherwise, returns true and the found version.
+func GetLatestVersionForMinorAutoUpdate(versions []gardencorev1beta1.ExpirableVersion, currentVersion string) (bool, string, error) {
+	// always first check if there is a higher patch version available
+	found, version, err := GetLatestVersionForPatchAutoUpdate(versions, currentVersion)
+	if found {
+		return found, version, nil
+	}
```

Comment:
Hi, that's because each update strategy first updates to the latest patch version before updating  to the auto-update target. That's how it is described in the issue [here](https://github.com/gardener/gardener/issues/8013) 

> always first try to update to the latest patch version of the same minor (essentially the auto-update for strategy patch).

Are you not ok with 
- the functionality / the update to the latest patch
- the cohesiveness of the function (returning higher patch if available even though claims it is the minor update routine? (solvable by adding an additional comment)


> +1 minor version

Please note that it does not always do minor +1 (like in k8s updates), but instead updates to the next higher minor in the cloudprofile (could be +n)